source/game_sa/CFire.cpp

```diff
@@ -1,6 +1,179 @@
 #include "StdInc.h"
 
-void CFire::Extinguish()
-{
-    plugin::Call<0x5393F0>();
+void CFire::ExtinguishWithWater(float fWaterStrength) {
+    const float fOriginalStrength = m_fStrength;
+    m_fStrength -= fWaterStrength * CTimer::ms_fTimeStep / 50.0f;
+
+    /* Create particles */
+    CVector particlePos = m_vecPosition + CVector{
+        CGeneral::GetRandomNumberInRange(-1.28f, 1.28f),
+        CGeneral::GetRandomNumberInRange(-1.28f, 1.28f),
+        CGeneral::GetRandomNumberInRange(-0.64f, 0.64f)
+        /* Original code:
+        (float)((rand() % 256) - 128) / 100.0f,
+        (float)((rand() % 256) - 128) / 100.0f,
+        (float)((rand() % 256) - 128) / 200.0f
+        */
```

Comment:
Please recheck Z-axis. **GetRandomNumberInRange** for **int** and **float** has different closing range (exclusive and inclusive respectively)

```suggestion
        CGeneral::GetRandomNumberInRange(-128, 128) / 100.0f,
        CGeneral::GetRandomNumberInRange(-128, 128) / 100.0f,
        CGeneral::GetRandomNumberInRange(0, 256) / 200.0f
```

Reply:
The difference is negligible.
Not worth to care about IMHO.
Using `GetRandomNumberInRange` by itself has more error than it not being inclusive.
Also, keep in mind that these are just random values in the air, a difference of 0.001 units isn't noticable at all.
Just wanted to use `CGeneral` to make the code uniform.